api/jwt/jwt.go

```diff
@@ -35,8 +35,13 @@ func NewService() (*Service, error) {
 }
 
 // GenerateToken generates a new JWT token.
-func (service *Service) GenerateToken(data *portainer.TokenData) (string, error) {
-	expireToken := time.Now().Add(time.Hour * 8).Unix()
+func (service *Service) GenerateToken(data *portainer.TokenData, timeout string) (string, error) {
+	userSessionTimeout, err := time.ParseDuration(timeout)
```

Comment:
The service should be updated to inject the duration (and parse it at injection time) inside the constructor.

It should also be updated to add a new method to the interface to update the session timeout (so that it can be used by the Settings handler when updating the Settings).